### ISSUE TITLE
Unexpected Poor Forecast for Air Passengers

### ISSUE THREAD

**Issue by abdulfatir** [Time-MoE/Time-MoE#5]
Thank you for releasing the pretrained model! I was testing the model on some toy datasets and found that it generates an unexpectedly poor forecast for the Air Passengers dataset. I am not sure if I am doing something incorrectly. 

**Result**:
![image](https://github.com/user-attachments/assets/5232eeed-3679-45ed-90e1-434853931dce)

**Code**:

```py
import torch
import pandas as pd
from transformers import AutoModelForCausalLM
import matplotlib.pyplot as plt
import numpy as np

df = pd.read_csv(
    "https://raw.githubusercontent.com/AileenNielsen/TimeSeriesAnalysisWithPython/master/data/AirPassengers.csv"
)
seqs = torch.tensor(df["#Passengers"].values).float().unsqueeze(0)

model = AutoModelForCausalLM.from_pretrained(
    "Maple728/TimeMoE-50M",
    device_map="cpu",
    trust_remote_code=True,
)
mean, std = seqs.mean(dim=-1, keepdim=True), seqs.std(dim=-1, keepdim=True)
normed_seqs = (seqs - mean) / std

prediction_length = 24
output = model.generate(normed_seqs, max_new_tokens=prediction_length)
normed_predictions = output[:, -prediction_length:]

predictions = normed_predictions * std + mean

plt.figure(figsize=(8, 4))

forecast_index = np.arange(prediction_length) + len(df["#Passengers"])
plt.plot(df["#Passengers"], color="royalblue", label="historical data")
plt.plot(forecast_index, predictions.squeeze(), color="tomato", label="forecast")
plt.legend()
plt.grid()
plt.show()
```

**Comment by Maple728** [comment]
In our pretrained time-series dataset, most patterns are seasonal without a clear trend, making Time-MoE more effective in capturing these seasonal patterns.

Zero-shot forecasting with the foundational model is sensitive to contextual values. We are actively working to address this issue. We recommend fine-tuning the Time-MoE model before using it for forecasting, as our experiments show that even a few epochs of training can significantly enhance performance.

Alternatively, you can replace standard normalization with max-value normalization to achieve better forecasting in time-series scenarios with an obvious trend:

**Code:**
```python
import torch
import pandas as pd
from transformers import AutoModelForCausalLM
import matplotlib.pyplot as plt
import numpy as np

df = pd.read_csv(
    "https://raw.githubusercontent.com/AileenNielsen/TimeSeriesAnalysisWithPython/master/data/AirPassengers.csv"
)
seqs = torch.tensor(df["#Passengers"].values).float().unsqueeze(0)

model = AutoModelForCausalLM.from_pretrained(
    "Maple728/TimeMoE-50M",
    device_map="cpu",
    trust_remote_code=True,
)
# mean, std = seqs.mean(dim=-1, keepdim=True), seqs.std(dim=-1, keepdim=True)
# normed_seqs = (seqs - mean) / std
max_val = seqs.max(dim=-1, keepdim=True)[0]
normed_seqs = seqs / max_val

prediction_length = 24
output = model.generate(normed_seqs, max_new_tokens=prediction_length)
normed_predictions = output[:, -prediction_length:]

# predictions = normed_predictions * std + mean
predictions = normed_predictions * max_val

plt.figure(figsize=(8, 4))

forecast_index = np.arange(prediction_length) + len(df["#Passengers"])
plt.plot(df["#Passengers"], color="royalblue", label="historical data")
plt.plot(forecast_index, predictions.squeeze(), color="tomato", label="forecast")
plt.legend()
plt.grid()
plt.show()
```

**Result**:
<img width="814" alt="image" src="https://github.com/user-attachments/assets/16610bc5-6f22-46ba-aec8-bab890b1faaf">

**Comment by abdulfatir** [comment]
@Maple728 Thanks for your quick reply. In general one cannot always know if the data will have trends. In this case, would you recommend a heuristic to use the "correct" normalization method?

**Comment by Maple728** [comment]
We generally recommend using standard normalization. If forecasting performance is not ideal, fine-tuning Time-MoE is a better option.

If you prefer not to fine-tune, consider this when choosing a normalization method:

- If after standard normalization, older observation values are mostly below 0, recent ones are above 0, and there's noticeable growth, max-value normalization might be preferable. (However, we can't guarantee it's always better than standard normalization.)

The normalization method doesn't fundamentally impact the zero-shot performance of Time-MoE; rather, it's the sensitivity to the scale of observation values that matters. We are actively working on this issue.

We hope this information is helpful to you.

**Comment by abdulfatir** [comment]
Very helpful, thanks!

**Comment by iganggang** [comment]
i feel your code is the same. why are the results different? i even ask gpt, it says you are same.

**Comment by Maple728** [comment]
> i feel your code is the same. why are the results different? i even ask gpt, it says you are same. Would you please tell me?

**Change Normalization Method from `zero` to `max`:**

```python
# Mean normalization method
mean, std = seqs.mean(dim=-1, keepdim=True), seqs.std(dim=-1, keepdim=True)
normed_seqs = (seqs - mean) / std

# Max normalization method
max_val = seqs.max(dim=-1, keepdim=True)[0]
normed_seqs = seqs / max_val
```

I hope you find this helpful!

**Comment by iganggang** [comment]
Thank you very much. I found it.
